### DISCUSSION TITLE
jotai complex modal state?

### DISCUSSION THREAD

**Discussion by ramunecroft** [pmndrs/jotai#2081] [Q&A] [answered]
In Jotai, when I have four modal windows named a, b, c, and d, I want to set a, b, and c to false when d is open. 
My current logic is too messy. have a better solution?. . .
 was able to apply it by creating a hook in the format of 'useXXX', but it seemed like the wrong approach😭

**Comment by ramunecroft** [comment]
and. additional 
I've written the code as shown below, but even though 'set' is defined, why isn't the state changing?

```
export const modalAState = atom(
  get => {
    const otherValue = get(modalDState);
    if (otherValue) return false;
    return true;
  },

  (get, set, newValue) => {
    set(modalAState, newValue);
  }
);

const [state, setState] = useAtom(modalAState);
  setState(true); // atom not writable
```

**Comment by dai-shi** [comment, marked as answer]
I think you should have one base atom to express which modal is open and you can define multiple derived atoms.

**Reply by ramunecroft** [reply]
I'll try 
thanks😸

**Reply by ramunecroft** [reply]
btw https://tutorial.jotai.org/ <- very helped me

**Reply by dai-shi** [reply]
@vasucp1207
